### PR TITLE
Updated the Render page to show errors at the top of the page

### PR DESCRIPTION
**Pull request by KietTruongSFDO**
# Critical Changes

# Changes
The render page has been updated to show errors that occur from saving, for example a validation rule. 
# Issues Closed

# New Metadata

# Deleted Metadata

### CODE REVIEW

File: src/lwc/utilPageLevelMessage/utilPageLevelMessage.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+    get subtitleText() {
+        return this.subtitle ? this.subtitle : '';
+    }
+
+    get hasIconName() {
+        return this.iconName ? true : false;
+    }
+
+    get assistiveText() {
```
Comment (bdvorachek) [inline src/lwc/utilPageLevelMessage/utilPageLevelMessage.js]: At first glance this looks odd, but are we looking to eventually return something other than SUCCESS for SUCCESS and so on in the future? Maybe we can set up an assistive text map variable?

File: src/lwc/geFormRenderer/geFormRenderer.js [hunk 2]
```diff
@@ -86,4 +152,40 @@ export default class GeFormRenderer extends NavigationMixin(LightningElement) {
     toggleSpinner() {
         this.showSpinner = !this.showSpinner;
     }
+
+    getDisplayedFieldsMappedByAPIName(sectionsList) {
+        let allFields = {};
+        sectionsList.forEach(section => {
+            const fields = section.getAllFieldsByAPIName();
+
+            allFields = Object.assign(allFields, fields);
```
Comment (bdvorachek) [inline src/lwc/geFormRenderer/geFormRenderer.js]: no need to change but FYI this is the same as `allFields = {...allFields, ...fields };`

File: src/lwc/geFormRenderer/geFormRenderer.js [hunk 3]
```diff
@@ -54,7 +57,70 @@ export default class GeFormRenderer extends NavigationMixin(LightningElement) {
...
             this.navigateToRecordPage(opportunityId);
-        });
+        })
+        .catch(error => {
+
+            this.toggleSpinner();
+            
+            // Show on top if it is a page level
+            this.hasPageLevelError = true;
+            let exceptionWrapper = JSON.parse(error.body.message);
```
Comment (bdvorachek) [inline src/lwc/geFormRenderer/geFormRenderer.js]: change to const

File: src/lwc/geFormRenderer/geFormRenderer.js [hunk 4]
```diff
@@ -86,4 +152,40 @@ export default class GeFormRenderer extends NavigationMixin(LightningElement) {
...
+    }
+
+    clearErrors() {
+
+        // Clear the page level error
+        this.pageLevelErrorMessage = '';
+
+        // Clear the field level errors
+        if (this.erroredFields.length > 0) {
+            for (var field in this.erroredFields) {
```
Comment (KietTruongSFDO) [inline src/lwc/geFormRenderer/geFormRenderer.js]: switch to .foreach

File: src/classes/ERR_ExceptionData.cls [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    }
+
+    public static String createExceptionWrapperJSONString(String errorMessage) {
+        ERR_ExceptionData exceptionWrapper = new ERR_ExceptionData(errorMessage);
+        String JSONExceptionData = JSON.serialize(exceptionWrapper);
+
+        return JSONExceptionData;
+    }
+
+    public static String createExceptionWrapperJSONString(Exception e) {
```
Comment (vmartinov-sforg) [inline src/classes/ERR_ExceptionData.cls]: Add a header comment here for the public method.

File: src/lwc/geFormRenderer/geFormRenderer.js (hunk 4 above)
Comment (vmartinov-sforg) [inline src/lwc/geFormRenderer/geFormRenderer.js]: Avoid using var, I think you can use const here.

File: src/lwc/geFormRenderer/geFormRenderer.js [hunk 6]
```diff
@@ -86,4 +152,40 @@ export default class GeFormRenderer extends NavigationMixin(LightningElement) {
...
+                var fieldToReset = this.erroredFields[field];
```
Comment (vmartinov-sforg) [inline src/lwc/geFormRenderer/geFormRenderer.js]: use const

File: src/classes/ERR_ExceptionData.cls [hunk 7]
```diff
@@ -0,0 +1,52 @@
+public with sharing class ERR_ExceptionData {
```
[the remark this answers is not in the document]
Comment (KietTruongSFDO) [reply src/classes/ERR_ExceptionData.cls]: Added

File: src/classes/ERR_ExceptionData.cls (hunk 5 above)
Comment (KietTruongSFDO) [reply src/classes/ERR_ExceptionData.cls]: Added

File: src/classes/ERR_ExceptionData.cls [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+        } else {
+            errorMessage = e.getMessage();
+        }
+    }
+
+    public ERR_ExceptionData(String errorMessage) {
+        this.errorMessage = errorMessage;
+    }
+
+    public static String createExceptionWrapperJSONString(String errorMessage) {
```
[the remark this answers is not in the document]
Comment (KietTruongSFDO) [reply src/classes/ERR_ExceptionData.cls]: Added

**Comment by markross-sfdo** [comment]
@KietTruongSFDO Hey what's the work item for this PR?

**Comment by KietTruongSFDO** [comment]
**lurch: attach W-037199

**Comment by LurchTheButler** [comment]
Tracking <a href="https://foundation.my.salesforce.com/a2x1E000001Ho5kQAC" target="_blank">W-037199</a>

**Comment by KietTruongSFDO** [comment]
@markross-sfdo  Just lurched it. Sorry!

**Pull request merged**